src/cpp/RiderLink/Source/RiderLoggingExtension/Private/RiderLoggingExtension.cpp

```diff
@@ -104,37 +119,27 @@ void FRiderLoggingExtensionModule::StartupModule()
 			{
 				DateTime = GetTimeNow(Time.GetValue());
 			}
-			const FString Msg = FString(msg);
 			const FString PlainName = Name.GetPlainNameString();
-
 			const JetBrains::EditorPlugin::LogMessageInfo MessageInfo{Type, PlainName, DateTime};
-
-			// [HACK]: fix https://github.com/JetBrains/UnrealLink/issues/17
-			// while we won't change BP hyperlink parsing
-			FString Tail = Msg.Left(4096);
-
-			FString ToSend;
-			while (Tail.Split("\n", &ToSend, &Tail))
+			LoggingScheduler->queue([Msg = FString(msg), MessageInfo]() mutable 
 			{
-				ToSend.TrimEndInline();
-				LoggingExtensionImpl::SendMessageToRider(MessageInfo, ToSend);
-			}
-			
-			Tail.TrimEndInline();            
-			LoggingExtensionImpl::SendMessageToRider(MessageInfo, Tail);
-        }
-    );
+				LoggingExtensionImpl::ScheduledSendMessage(&Msg, MessageInfo);
+			});
+		}
+		);
+	},
+	[this]()
+	{
+		if(OutputDevice.onSerializeMessage.IsBound())
+			OutputDevice.onSerializeMessage.Unbind();
+	});
```

Comment:
don't we need to reset scheduler explicily here?

Reply:
it'll be killed with the same lifetime